### PR TITLE
Release Notes for March 2024 Prerelease SDK

### PR DESCRIPTION
**Pull request by mikehoffms**
Rendered article section for review:
https://review.learn.microsoft.com/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3115#102470-prerelease

This PR replaces https://github.com/MicrosoftDocs/edge-developer/pull/3095 which was created prior to replacing relnotes .md file by 3 new files.

AB#49613562

### CODE REVIEW

File: microsoft-edge/webview2/release-notes/index.md [hunk 1]
```diff
@@ -85,6 +85,239 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+* [CoreWebView2FileSystemHandleKind Enum](/dotnet/api/microsoft.web.webview2.core.corewebview2filesystemhandlekind?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
+   * `File`
+   * `Directory`
+
+* [CoreWebView2FileSystemHandlePermission Enum](/dotnet/api/microsoft.web.webview2.core.corewebview2filesystemhandlepermission?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
+   * `ReadOnly`
+   * `ReadWrite`
+
+* `CoreWebView2ObjectCollection` Class:
+    * [CoreWebView2ObjectCollection.InsertValueAtIndex Property] - todo: link
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes/index.md:159]: link .NET CoreWebView2ObjectCollection.InsertValueAtIndex & .RemoveValueAtIndex

File: microsoft-edge/webview2/release-notes/index.md [hunk 2]
```diff
@@ -85,6 +85,237 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+   * [ICoreWebView2ExperimentalFileSystemHandle::get_Path](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?view=webview2-1.0.2470-prerelease&preserve-view=true#get_path)
+   * [ICoreWebView2ExperimentalFileSystemHandle::get_Permission](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?view=webview2-1.0.2470-prerelease&preserve-view=true#get_permission)
+
+* [ICoreWebView2ExperimentalObjectCollection](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true)
+   * [ICoreWebView2ExperimentalObjectCollection::get_Kind](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true#get_kind)<!-- todo: delete?  there's no "kind" member at https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-win32-preview&branch=pr-en-us-87 or in .net or rt tab above-->
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes/index.md:202]: delete item? there's no "kind" member at https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-win32-preview&branch=pr-en-us-87 or in .net or rt tab above
@peiche-jessica @victorhuangwq

File: microsoft-edge/webview2/release-notes/index.md [hunk 3]
```diff
@@ -85,6 +85,237 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+
+###### Runtime-only
+
+* Fixed a reliability regression that could crash the application process when an old version of WebView2 client DLL is unloaded.
+
+
+###### Runtime and SDK
+
+* Ensured that the WebView2 temporary download folder is unique per user data folder, and doesn't interfere with other apps or the browser.
+<!-- todo: confirm that this is both Runtime & SDK.  vs. the orig commit that added this item in both the "Runtime-only" and "SDK-only" sections -->
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes/index.md:313]: confirm that this is both Runtime & SDK. vs. the orig [commit](https://github.com/MicrosoftDocs/edge-developer/pull/3095/commits/ae1229a86d2f2fee92b51ee1f48568966fd14c70) that added this item in both the "Runtime-only" and "SDK-only" sections:
image
@peiche-jessica @victorhuangwq

Comment (peiche-jessica) [reply microsoft-edge/webview2/release-notes/index.md:313]: I checked the original work item. This should be runtime-only. Item owner is Diana if we'd like triple check.

File: microsoft-edge/webview2/release-notes/index.md [hunk 4]
```diff
@@ -85,6 +85,238 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+   * [CoreWebView2Environment.CreateWebFileSystemFileHandle Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#createwebfilesystemfilehandle)
+
+* `CoreWebView2FileSystemHandle` Class:
+   * [CoreWebView2FileSystemHandle.Kind Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2filesystemhandle?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#kind)
+   * [CoreWebView2FileSystemHandle.Permission Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2filesystemhandle?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#permission)
+   <!-- todo: add Path item, like Win32 & .net tab?  https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?branch=pr-en-us-87&view=webview2-win32-preview -->
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes/index.md:174]: add a "CoreWebView2FileSystemHandle.Path Property" item, like Win32 & .net tab? 

Win32 topic has get_Path:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?branch=pr-en-us-87&view=webview2-win32-preview

WinRT topic lacks Path:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2filesystemhandlekind?view=webview2-winrt-preview&branch=pr-en-us-55

@peiche-jessica @victorhuangwq

Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes/index.md:174]: added Path in WinRT:
```
* `CoreWebView2FileSystemHandle` Class:
   * [CoreWebView2FileSystemHandle.Kind Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2filesystemhandle?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#kind)
   * [CoreWebView2FileSystemHandle.Path Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2filesystemhandle?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#path)
   * [CoreWebView2FileSystemHandle.Permission Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2filesystemhandle?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#permission)
```

File: microsoft-edge/webview2/release-notes/index.md (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes/index.md:314]: image

File: microsoft-edge/webview2/release-notes/index.md (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes/index.md]:
```
* [ICoreWebView2ExperimentalFileSystemHandle](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?view=webview2-1.0.2470-prerelease&preserve-view=true)
   * [ICoreWebView2ExperimentalFileSystemHandle::get_Kind](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?view=webview2-1.0.2470-prerelease&preserve-view=true#get_kind)
   * [ICoreWebView2ExperimentalFileSystemHandle::get_Path](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?view=webview2-1.0.2470-prerelease&preserve-view=true#get_path)
   * [ICoreWebView2ExperimentalFileSystemHandle::get_Permission](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalfilesystemhandle?view=webview2-1.0.2470-prerelease&preserve-view=true#get_permission)

* [ICoreWebView2ExperimentalObjectCollection](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true)
   * [ICoreWebView2ExperimentalObjectCollection::InsertValueAtIndex](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true#insertvalueatindex)
   * [ICoreWebView2ExperimentalObjectCollection::RemoveValueAtIndex](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true#removevalueatindex)
```

Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes/index.md]: the 3 platform tabs now list the Kind, Path, & Permission members for FileSystemHandle

File: microsoft-edge/webview2/release-notes/index.md (hunk 1 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes/index.md]: manually constructed the links:
```
* `CoreWebView2ObjectCollection` Class:
   * [CoreWebView2ObjectCollection.InsertValueAtIndex Property](/dotnet/api/microsoft.web.webview2.core.corewebview2objectcollection.insertvalueatindex?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
   * [CoreWebView2ObjectCollection.RemoveValueAtIndex Property](/dotnet/api/microsoft.web.webview2.core.corewebview2objectcollection.removevalueatindex?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
```

**Review by nishitha-burman — APPROVED** [review]

File: microsoft-edge/webview2/release-notes/index.md [hunk 5]
```diff
@@ -85,6 +85,232 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+   * [ICoreWebView2ExperimentalEnvironmentOptions2::get_ScrollBarStyle](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalenvironmentoptions2?view=webview2-1.0.2470-prerelease&preserve-view=true#get_scrollbarstyle)
+   * [ICoreWebView2ExperimentalEnvironmentOptions2::put_ScrollBarStyle](/microsoft-edge/webview2/reference/win32/icorewebview2experimentalenvironmentoptions2?view=webview2-1.0.2470-prerelease&preserve-view=true#put_scrollbarstyle)
+
+* [COREWEBVIEW2_SCROLLBAR_STYLE enum](/microsoft-edge/webview2/reference/win32/webview2experimental-idl?view=webview2-1.0.2470-prerelease#corewebview2_scrollbar_style)
+
+---
+
+
+<!-- ---------- -->
+* Updated the WebMessageObjects API to allow injecting DOM objects into WebView2 content that's constructed via the app and via the `CoreWebView2.PostWebMessage` API in the other direction.  Added a new web object type to represent a file system handle.
```
Comment (victorhuangwq) [inline microsoft-edge/webview2/release-notes/index.md:134]:
```suggestion
* Updated the WebMessageObjects API to allow injecting DOM objects into WebView2 content constructed via the app and via the `CoreWebView2.PostWebMessage` API in the other direction.  Added a new web object type to represent a file system handle that can be posted to the web content to provide it with filesystem access.
```

The dev owner of this API suggested this change to clarify the purpose of the file system handle.

**Review by victorhuangwq — APPROVED** [review]
LGTM

**Review by peiche-jessica — APPROVED** [review]

File: microsoft-edge/webview2/release-notes/index.md [hunk 6]
```diff
@@ -85,6 +85,232 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+* [CoreWebView2FileSystemHandleKind Enum](/dotnet/api/microsoft.web.webview2.core.corewebview2filesystemhandlekind?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
+   * `File`
+   * `Directory`
+
+* [CoreWebView2FileSystemHandlePermission Enum](/dotnet/api/microsoft.web.webview2.core.corewebview2filesystemhandlepermission?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
+   * `ReadOnly`
+   * `ReadWrite`
+
+* `CoreWebView2ObjectCollection` Class:
+   * [CoreWebView2ObjectCollection.InsertValueAtIndex Property](/dotnet/api/microsoft.web.webview2.core.corewebview2objectcollection.insertvalueatindex?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes/index.md:159]: not found for .NET:
* `CoreWebView2ObjectCollection` Class:
   * [CoreWebView2ObjectCollection.InsertValueAtIndex Property](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2objectcollection.insertvalueatindex?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)
   * [CoreWebView2ObjectCollection.RemoveValueAtIndex Property](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2objectcollection.removevalueatindex?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true)

eg near: 
https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2filesystemhandlepermission?view=webview2-dotnet-1.0.2470-prerelease&preserve-view=true

They are found for other platforms: 

WinRT:
* `CoreWebView2ObjectCollection` Class:
   * [CoreWebView2ObjectCollection.InsertValueAtIndex Property](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2objectcollection?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#insertvalueatindex)
   * [CoreWebView2ObjectCollection.RemoveValueAtIndex Property](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2objectcollection?view=webview2-winrt-1.0.2470-prerelease&preserve-view=true#removevalueatindex)

Win32:
* [ICoreWebView2ExperimentalObjectCollection](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true)
   * [ICoreWebView2ExperimentalObjectCollection::InsertValueAtIndex](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true#insertvalueatindex)
   * [ICoreWebView2ExperimentalObjectCollection::RemoveValueAtIndex](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2experimentalobjectcollection?view=webview2-1.0.2470-prerelease&preserve-view=true#removevalueatindex)

@peiche-jessica @victorhuangwq

Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes/index.md]: kept for win32 only

**Comment by mikehoffms** [comment]
Closing PR w/o merging, to avoid GitHub bug/limitation where it claims there's a "merge conflict" and tries to interleave the Release & Prerelease sections.  Copied the final, approved, tested h2 Prerelease section from the present PR to PR https://github.com/MicrosoftDocs/edge-developer/pull/3114.

**Pull request closed without merging**